migration/util.go

```diff
@@ -204,3 +205,45 @@ func CachedMigration(cache MigrationCache, m ActorMigration) ActorMigration {
 		cache:          cache,
 	}
 }
+
+type DeferredMigrationSet struct {
+	todo map[address.Address]struct{}
+	lk   sync.Mutex
+}
+
+func NewDeferredMigrationSet() *DeferredMigrationSet {
+	return &DeferredMigrationSet{
+		todo: make(map[address.Address]struct{}),
+	}
+}
+
+type DeferredMigrator struct {
+	ds  *DeferredMigrationSet
+	sub ActorMigration
+}
+
+func NewDeferredMigrator(ds *DeferredMigrationSet, sub ActorMigration) *DeferredMigrator {
+	return &DeferredMigrator{
+		ds:  ds,
+		sub: sub,
+	}
+}
+
+func (d DeferredMigrator) MigratedCodeCID() cid.Cid {
+	panic("this can't be called")
+}
+
+var errMigrationDeferred = errors.New("migration deferred")
+
+func (d DeferredMigrator) MigrateState(ctx context.Context, store cbor.IpldStore, in ActorMigrationInput) (*ActorMigrationResult, error) {
+	d.ds.lk.Lock()
+	defer d.ds.lk.Unlock()
+
+	_, ok := d.ds.todo[in.Address]
+	if ok {
+		return nil, xerrors.Errorf("actor %s already in deferred set", in.Address)
+	}
+	d.ds.todo[in.Address] = struct{}{}
+
+	return nil, errMigrationDeferred
```

Comment:
In #240 to avoid rebasing - https://github.com/filecoin-project/go-state-types/pull/240/commits/7e39f88cd964d598a558f382075dd48a9c401003

(maybe makes sense to just merge 240 here now, but separate is probably easier to review)

Reply:
Can we just move the entire migration to that PR? If we can't make changes to this PR, the review process is kind of broken. That and the other PR makes _so_ many changes to the migration that I'm not sure if there's even a point of reviewing it here.

Reply:
Everything is here now